elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,537 @@
+                ((Field) fieldSelection).getSelectionSet().getSelections().forEach(
+                        selection -> addSelection(selection, parentProjection));
+            } else {
+                addField((Field) fieldSelection, parentProjection);
+            }
+        } else {
+            throw new InvalidEntityBodyException(
+                    String.format("Unsupported selection type {%s}.", fieldSelection.getClass()));
+        }
+    }
+
+    /**
+     * Resolve a graphQL {@link FragmentSpread} into {@link Selection}s and add them to an {@link EntityProjection}
+     *
+     * @param fragment graphQL fragment
+     * @param parentProjection entity projection that contains this fragment
+     */
+    private void addFragment(FragmentSpread fragment, EntityProjection parentProjection) {
+        String fragmentName = fragment.getName();
+        if (!fragmentResolver.contains(fragmentName)) {
+            throw new InvalidEntityBodyException(String.format("Unknown fragment {%s}.", fragmentName));
+        }
+
+        FragmentDefinition fragmentDefinition = fragmentResolver.get(fragmentName);
+
+        // type name in type condition of the Fragment must match the entity projection type name
+        if (entityDictionary.getJsonAliasFor(parentProjection.getType())
+                .equals(fragmentDefinition.getTypeCondition().getName())) {
+            fragmentDefinition.getSelectionSet().getSelections()
+                    .forEach(selection -> addSelection(selection, parentProjection));
+        }
+    }
+
+    /**
+     * Add a new graphQL {@link Field} into an {@link EntityProjection}
+     *
+     * @param field graphQL field
+     * @param parentProjection entity projection that contains this field
+     */
+    private void addField(Field field, EntityProjection parentProjection) {
+        Class<?> parentType = parentProjection.getType();
+        String fieldName = field.getName();
+
+        // this field would either be a relationship field or an attribute field
+        if (entityDictionary.getRelationshipType(parentType, fieldName) != RelationshipType.NONE) {
+            // handle the case of a relationship field
+            final Class<?> relationshipType = entityDictionary.getParameterizedType(parentType, fieldName);
+
+            // build new entity projection with only entity type and entity dictionary
+            EntityProjection relationshipProjection = createProjection(relationshipType, field);
+            Relationship relationship = Relationship.builder()
+                    .name(fieldName)
+                    .alias(field.getAlias())
+                    .projection(relationshipProjection)
+                    .build();
+
+            relationshipMap.put(field.getSourceLocation(), relationship);
+
+            // add this relationship projection to its parent projection
+            parentProjection.getRelationships().add(relationship);
```

Comment:
Projections are really intended to be immutable.  If you want to edit a projection, you should be dealing with a Builder - not the projection itself.